### PR TITLE
Post Normalizer: Refactor rules into individual files

### PR DESCRIPTION
**Pull request by blowery**
And make the sync rules truly sync, wrapping them in the main post normalizer.

This really just moves all of the post normalizer rules out of the one massive file into little files. This helps scoping quite a bit and makes it clearer what each rule does and what constants it depends on.

I also took the opportunity to make all of the sync rules look and feel like sync functions. No longer do they take a callback, which implies async behavior, and invoke is synchronously. There is a new wrapper that converts the sync variants to look like the old async-ish version so the refactoring doesn't impact users of the post normalizer. 

To test, 
1. Run the test suite. Yay tests!
2. Try using the Reader. Posts should categorize based on content, like they do now. Posts without a featued image spec'd should get one when a good candidate is found
3. Try using My Posts. Same basic set of criteria.

Stuff for another PR
- [ ] Rework lib/feed-post-store/* to use the rules directly and avoid using postNormalizer and async
- [ ] Rework lib/posts to use the rules directly
- [ ] Find anywhere else that's using post norm and rework it
- [ ] Kill post norm and the async dependency

### CODE REVIEW

**Comment by blowery** [comment]
Tied to #5489 and #5564

**Comment by blowery** [comment]
Stuff left to do:
- [] Rework lib/feed-post-store/* to use the rules directly and avoid using postNormalizer and async
-[] Rework lib/posts to use the rules directly
-[] Find anywhere else that's using post norm and rework it
-[] Kill post norm and the async dependency

**Comment by blowery** [comment]
cc @aduth @rralian @bluefuton @TooTallNate

**Comment by rralian** [comment]
tl;dr > looks good to 🚢  to me.

-------------------------------------

The tests all pass, and I've run through the reader, post-list, and page-list pages, switched sites, and tried loading with empty localstorage and see no new issues or regressions.

I'm not totally sure how to test this bit.

> 2. Try using the Reader. Posts should categorize based on content, like they do now. Posts without a featued image spec'd should get one when a good candidate is found

I can confirm that posts without a featured image are getting an image when a good candidate is found, but I'm not sure how else to test post categoziation, because I'm not familiar with the view for different types of posts. I did however check the classes applied to the article to make sure they had relevant classes added. For the most part that was correct. however I did see an example of a post that got "is-one-liner" applied when it was clearly not a one-liner. Your call whether this is an issue, but the display seems fine.

one-liner

I also noticed this issue when loading the post-list page (`/posts/:site`) with a clear localStorage, and I got this error. Just letting you know this appears to be an error in master and _not_ something you should have to worry about. I'll create a separate issue to track it.

empty localstorage error
